core/arch/arm/kernel/thread_a64.S

```diff
@@ -42,6 +42,8 @@ FUNC thread_resume , :
 	load_xregs x0, THREAD_CTX_REGS_X1, 1, 3
 	ldr	x0, [x0, THREAD_CTX_REGS_X0]
 	eret
+	dsb nsh
```

Comment:
This deserves at least a comment describing what's going on.

However, I wonder if it wouldn't be preferable to wrap this in a local macro instead. How about something like:
``` asm
.macro return_from_exception
    eret
    /* Guard against speculation (feel free to elaborate...)*/
    dsb nsh
    isb
.endm
```

 I see that the Linux implementation uses the `sb` instruction, while the BSD implementation relies on the `dsb`, `isb` sequence. I suppose it doesn't matter much.

Reply:
Done. sb is not an instruction:
https://github.com/torvalds/linux/blob/cef7298262e9af841fb70d8673af45caf55300a1/arch/arm64/include/asm/assembler.h#L116

Reply:
Aha, thanks for the link.

Reply:
Can you please merge this? I don't have write access to the repository.